### ISSUE TITLE
Building and running samples error (probably path related)

### ISSUE THREAD

**Issue by noncom** [zig-gamedev/zig-gamedev#511]
I just cloned the repo, and I can't run the first recommended example.
When trying to run it I'm getting this:

```powershell
Microsoft Windows [Version 10.0.19045.4046]
(c) Microsoft Corporation. All rights reserved.

D:\work\zig\zig-gamedev>zig build physically_based_rendering_wgpu-run
physically_based_rendering_wgpu-run
└─ run physically_based_rendering_wgpu
   └─ physically_based_rendering_wgpu
      └─ install physically_based_rendering_wgpu
         └─ zig build-exe physically_based_rendering_wgpu Debug native
            └─ zig build-lib zmesh Debug native 1 errors
error: unable to check cache: stat file 'D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs\zmesh\libs\meshoptimizer\clusterizer.cpp' failed: Unexpected     
error: the following command failed with 1 compilation errors:
D:\storage\distribs\programming\__zig\zig-windows-x86_64-0.12.0-dev.3074+ae7f3fc36\zig.exe build-lib -cflags -std=c99 -fno-sanitize=undefined -DPAR_SHAPES_T=uint32_t -- D:\work\zig\zig-gamedev\libs\zmesh\libs\par_shapes\par_shapes.c -cflags  -- D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/clusterizer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/indexgenerator.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/vcacheoptimizer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/vcacheanalyzer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/vfetchoptimizer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/vfetchanalyzer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/overdrawoptimizer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/overdrawanalyzer.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/simplifier.cpp D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs/zmesh/libs/meshoptimizer/allocator.cpp -cflags -std=c99 -- D:\work\zig\zig-gamedev\libs\zmesh\libs\cgltf\cgltf.c -ODebug -I D:\work\zig\zig-gamedev\libs\zmesh\libs\par_shapes -I D:\work\zig\zig-gamedev\libs\zmesh\libs\cgltf 
-Mroot -lc++ -lc --cache-dir D:\work\zig\zig-gamedev\zig-cache --global-cache-dir C:\Users\user\AppData\Local\zig --name zmesh -static --listen=- 
physically_based_rendering_wgpu-run
└─ run physically_based_rendering_wgpu
   └─ physically_based_rendering_wgpu
      └─ install physically_based_rendering_wgpu
         └─ zig build-exe physically_based_rendering_wgpu Debug native
            └─ zig build-lib zgui Debug native 1 errors
error: unable to check cache: stat file 'D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs\zgui\libs\imgui\imgui.cpp' failed: Unexpected

... a lot of similar output ...
```

Where the problem I'm seeing is that the paths it's trying to work with are not correct, like:

```
D:\work\zig\zig-gamedev\D:\work\zig\zig-gamedev\libs\zmesh\libs\meshoptimizer\clusterizer.cpp
```

And any other example I try to run fails with the same error.
Could it be some issue with running it on Windows?

**Comment by hazeycode** [comment]
I've seen this happen on both windows and aarch64-macOS. It can be worked around by deleting the zig-cache folder every time, it's very annoying.

**Comment by bgourlie** [comment]
I believe this is a regression in the nightly builds of zig. Using the mach nominated build of zig 0.12 fixed it for me.

**Comment by hazeycode** [comment]
> Here is a discussion about this issue: https://github.com/ziglang/zig/pull/19017#issuecomment-1969232794
> 
> 
> 
> I believe the solution is to use b.dependency instead of the current `Package.link` method.
> 
> 
> 
> I'm happy to implement this change, but it would be breaking.

I've been thinking about making this change for other reasons. So maybe it's time to bite the bullet and go for it if doing so also resolves this issue.

I think it can be done without breaking everyone's build scripts if we also keep the package API around for now. Although some tweaks are required to each libs build.zig in order to do so, namely in how runTests is defined and invoked.

**Comment by kcbanner** [comment]
That makes sense, I can get started on this today unless you had a specific plan?

**Comment by hazeycode** [comment]
It would be a big help, thanks!

**Comment by kcbanner** [comment]
I opened a PR with just the zmesh changes here: https://github.com/zig-gamedev/zig-gamedev/pull/514, but it's ideal form is blocked on some build system changes.

**Comment by hazeycode** [comment]
@Angluca we do not provide a Dawn binary for macOS 11. I'm not sure if we will be able to (building Dawn is a huge PITA) but please open a separate issue for this.

**Comment by hazeycode** [comment]
Fixed by https://github.com/zig-gamedev/zig-gamedev/pull/518
